libs/wingsdk/src/target-sim/bucket.inflight.ts

```diff
@@ -136,14 +137,30 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
     });
   }
 
-  public async get(key: string): Promise<string> {
+  public async get(key: string, options?: BucketGetOptions): Promise<string> {
     return this.context.withTrace({
       message: `Get (key=${key}).`,
       activity: async () => {
         const hash = this.hashKey(key);
         const filename = join(this._fileDir, hash);
         try {
-          return await fs.promises.readFile(filename, "utf8");
+          const file = await fs.promises.open(filename, "r");
+          const stat = await file.stat();
+
+          let start = 0;
+          if (options?.start !== undefined) {
+            start = options.start;
+          }
+
+          let length = stat.size;
+          if (options?.end !== undefined) {
+            length = options.end + 1;
+          }
+
+          const buffer = Buffer.alloc(length - start);
+          await file.read(buffer, 0, length - start, start);
+          await file.close();
+          return buffer.toString("utf8");
```

Comment:
Buffer will allow invalid UTF-8 strings to be parsed:

```js
Buffer.from([240, 160, 174, 183]).toString("utf8") // 𠮷
Buffer.from([240, 160, 174]).toString("utf8") // �
```

Would it be better to throw an error if the bytes are invalid? I believe TextDecoder would handle this for us:

```js
new TextDecoder("utf8", { fatal: true }).decode(new Uint8Array([240, 160, 174, 183])) // 𠮷
new TextDecoder("utf8", { fatal: true }).decode(new Uint8Array([240, 160, 174])) // error thrown
```